src/orion/core/io/config.py

```diff
@@ -451,3 +451,35 @@ def to_dict(self):
                 config[key] = self[key].to_dict()
 
         return config
+
+    def from_dict(self, config):
+        """Set the configuration from a dictionary"""
+
+        logger.debug("Setting config to %s", config)
+        logger.debug("Config was %s", repr(self))
+
+        with _disable_logger():
+            for key in self._config:  # pylint: disable=consider-using-dict-items
+                value = config.get(key)
+
+                if value is not None:
+                    self[key] = value
+                else:
+                    self._config[key].pop("value", None)
+
+            for key in self._subconfigs:
+                value = config.get(key)
+                self[key].from_dict(value)
+
+        logger.debug("Config is %s", repr(self))
+
+    def __repr__(self) -> str:
+        confs = []
+        for k, v in self._config.items():
+            if "value" in v:
+                confs.append(f'{k}: {repr(v["value"])}')
+
+        for k, v in self._subconfigs.items():
+            confs.append(f"{k}: {repr(v)}")
+
+        return "{" + ", ".join(confs) + "}"
```

Comment:
Would `pprint.pformat(self.to_dict())` do the job? I would require less testing.

Reply:
added